alacritty/src/event.rs

```diff
@@ -381,16 +384,54 @@ impl<N: Notify + OnResize> Processor<N> {
             config,
             message_buffer,
             display,
+            event_queue: Vec::new(),
         }
     }
 
+    /// Return `true` if `event_queue` is empty, `false` otherwise.
+    #[inline]
+    #[cfg(not(any(target_os = "macos", windows)))]
+    fn event_queue_empty(&mut self) -> bool {
+        let wayland_event_queue = match self.display.wayland_event_queue.as_mut() {
+            Some(wayland_event_queue) => wayland_event_queue,
+            None => return self.event_queue.is_empty(),
```

Comment:
This is quite an important little line. Could we add a line of documentation to it that says that on X11 we just check the event queue?

Suggestion:

```
// Since frame callbacks do not exist on X11, just check the event queue.
```